src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -766,4 +769,26 @@ public static boolean getMobGriefingEvent(World world, Entity entity)
         Result result = event.getResult();
         return result == Result.DEFAULT ? world.getGameRules().getBoolean("mobGriefing") : result == Result.ALLOW;
     }
+
+    /**
+     * Posts the world unload event.
+     * Optionally catches and suppresses any exceptions to allow the game to safely finish unloading the world.
+     */
+    public static boolean onWorldUnload(World world, boolean suppressExceptions)
+    {
+        WorldEvent.Unload event = new WorldEvent.Unload(world);
+        try
+        {
+            return MinecraftForge.EVENT_BUS.post(event);
+        }
+        catch (Throwable e)
+        {
+            LOGGER.error("A WorldEvent.Unload event handler has crashed. Please report this to the mod author, it is a very serious error. This may result in data loss because other unload event handlers could be skipped.", e);
```

Comment:
If it is that serious it should probably be fatal instead of error

Reply:
I thought about it, but fatal generally means that we cannot recover and the game will shut down immediately, and that is not really true here.

Reply:
Yeah maybe, but it is an exception that is fatal and fatal is also used in other places in forge where no crash occures (see https://github.com/MinecraftForge/MinecraftForge/blob/412c2ae53f8dd24a7ed26ee0783c3585e44522ba/src/main/java/net/minecraftforge/client/model/ModelLoader.java#L1059 ), but thats up to you